### PR TITLE
chore(wallet): migrate to Blowfish API v2023-06-05

### PR DESCRIPTION
**Pull request by onyb**
Resolves https://github.com/brave/brave-browser/issues/34619

| EVM docs | https://docs.blowfish.xyz/reference/scan-transactions-evm |
-|-

|Solana docs | https://docs.blowfish.xyz/reference/scan-transactions-solana|
-|-

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

N/A

### CODE REVIEW

File: components/brave_wallet/browser/simulation_service.cc [hunk 1]
```diff
@@ -150,9 +155,7 @@ GURL SimulationService::GetScanTransactionURL(const std::string& chain_id,
                                               const std::string& language) {
   std::string spec = base::StringPrintf(
       "%s/%s/%s", kBlowfishBaseAPIURL,
-      GetRelativeScanPath(chain_id, coin).c_str(),
-      coin == mojom::CoinType::SOL ? "transactions" : "transaction");
-
+      GetRelativeScanPath(chain_id, coin).c_str(), "transactions");
```
Comment (supermassive) [inline components/brave_wallet/browser/simulation_service.cc:158]: nit: can we have a DCHECK on supported coin arg?

File: components/brave_wallet/browser/simulation_responses.idl [hunk 2]
```diff
@@ -16,19 +16,26 @@ namespace simulation_responses {
...
-    MAINNET_REPLAY_POSSIBLE,
+    MALICIOUS_PACKAGES,
+    MAINNET_REPLAY_POSSIBLE, // TODO(onyb): DEPRECATED
     MULTI_COPY_CAT_DOMAIN,
     NEW_DOMAIN,
-    NON_ASCII_URL,
-    OBFUSCATED_CODE,
+    NON_ASCII_URL, // TODO(onyb): DEPRECATED
+    OBFUSCATED_CODE, // TODO(onyb): DEPRECATED
     PERMIT_NO_EXPIRATION,
```
Comment (supermassive) [inline components/brave_wallet/browser/simulation_responses.idl:39]: why we keep deprecated ones here?

File: components/brave_wallet/browser/simulation_response_parser.cc [hunk 3]
```diff
@@ -586,42 +669,51 @@ mojom::EVMSimulationResponsePtr ParseSimulationResponse(
...
+          .expected_state_changes.GetDict();
 
-    if (auto raw_info = ParseStateChangeRawInfo(state_change_value.raw_info)) {
-      state_change->raw_info = std::move(raw_info);
+  auto* account_state_changes_value =
+      expected_state_changes_value.FindList(user_account)
+          ?: expected_state_changes_value.FindList(
+                 base::ToLowerASCII(user_account));
+  if (!account_state_changes_value) {
+    return nullptr;
```
Comment (supermassive) [inline components/brave_wallet/browser/simulation_response_parser.cc:705]: Not sure what is `?:` here

File: components/brave_wallet/browser/simulation_responses.idl (hunk 2 above)
Comment (onyb) [reply components/brave_wallet/browser/simulation_responses.idl]: Thanks for pointing out. Removed them here: https://github.com/brave/brave-core/pull/21609/commits/1e230b692bde2fac459f5550e281ba3d2768a4c6

File: components/brave_wallet/browser/simulation_response_parser.cc (hunk 3 above)
Comment (onyb) [reply components/brave_wallet/browser/simulation_response_parser.cc]: Removed the Elvis operator: https://github.com/brave/brave-core/pull/21609/commits/fdb6f531bbd4b06b4cc2c7f24aca19622537070b

File: components/brave_wallet/browser/simulation_service.cc (hunk 1 above)
Comment (onyb) [reply components/brave_wallet/browser/simulation_service.cc:159]: Added: https://github.com/brave/brave-core/pull/21609/commits/99e1648dd202ccacd59eae2b593ff448be84a68d

**Review by supermassive — APPROVED** [review]
wallet core lgtm

**Review by supermassive — APPROVED** [review]
wallet core lgtm

**Review by josheleonard — APPROVED** [review]
Frontend ++

Great work!

**Review by StephenHeaps — APPROVED** [review]
iOS++

File: components/brave_wallet/browser/simulation_response_parser_unittest.cc [hunk 4]
```diff
@@ -1633,159 +1692,155 @@ TEST(SimulationResponseParserUnitTest, ParseSolanaWarnings) {
...
+                "kind": "SPL_TRANSFER",
+                "data": {
+                  "asset": {
+                    "symbol": "USDT",
+                    "name": "USDT",
+                    "mint": "Es9vMFrzaCERmJfrF4H2FYD4KCoNkY11McCe8BenwNYB",
+                    "decimals": "6",
+                    "supply": "1000000000",
+                    "metaplexTokenStandard": "unknown",
+                    "price": null,
```
Comment (cdesouza-chromium) [inline components/brave_wallet/browser/simulation_response_parser_unittest.cc:1720]: There was a bug on this relating to this field not being substituted. This was recently fixed in a separate PR [1]. A review of these tests should follow to make sure they cover both fail and success routes to make sure we are in general testing what we think to be testing. An [2] issue has been created for this review.

[1] https://github.com/brave/brave-core/pull/25643
[2] https://github.com/brave/brave-browser/issues/41144

**Pull request merged**